packages/styled-components/src/models/ComponentStyle.js

```diff
@@ -81,7 +80,7 @@ export default class ComponentStyle {
       const name = generateName(dynamicHash >>> 0);
 
       if (!styleSheet.hasNameForId(componentId, name)) {
-        const cssFormatted = stringifyRules(css, `.${name}`, undefined, componentId);
+        const cssFormatted = styleSheet.stringifier(css, `.${name}`, undefined, componentId);
```

Comment:
So, this might be a nit, but it'd seem like `stringifier` couples the new Sheet implementation to Stylis technically, or at least to some kind of `stylis`-like stringifier even if it's refactored, which makes the Sheet stuff on its own less reusable for others, once we make it a separate package.

Does it make sense to have an intermediate object that holds the sheet and the stringifier separately in the `StyleSheetManager` context?

Reply:
It would end up being a much bigger change... but also the stringification process is pretty closely linked with stylesheet injection so I don't think the coupling is actually a bad thing.

Reply:
I mean, it's not a bad thing, but it will make it hard to extract `styled-sheet` without coupling it with stylis. We can always make this change to enable such an extraction separately though.